packages/transactions/src/builders.ts

```diff
@@ -1692,3 +1723,70 @@ export async function estimateTransactionFeeWithFallback(
     throw error;
   }
 }
+
+/** @internal multi-sig signing re-use */
+function mutatingSignAppendMultiSig(
+  /** **Warning:** method mutates `transaction` */
+  transaction: StacksTransaction,
+  publicKeys: string[],
+  signerKeys: string[],
+  address?: string
+) {
+  if (isSingleSig(transaction.auth.spendingCondition)) {
+    throw new Error('Transaction is not a multi-sig transaction');
+  }
+
+  const signer = new TransactionSigner(transaction);
+
+  const pubs = ensureValidMultiSigPublicKeyOrder(
+    publicKeys,
+    transaction.auth.spendingCondition.signaturesRequired,
+    transaction.auth.spendingCondition.hashMode,
+    address ? createAddress(address).hash160 : undefined
```

Comment:
The ternary operator here is kinda pointless, it would just map any "falsy" value like `0`, `false`, or `null` to `undefined`

Reply:
Removed the previous optionality of this param, but kept the ternary (that's fine in this case, right? We want falsy values not to be treated as an address. We typically don't strictly check each param type and trust most developers will have JS/TS tooling to disallow most types. At least that's a common pattern in the codebase.)